packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -64,6 +72,76 @@ const branchProtectionDefaults = deepFreeze({
  * Main.  On a nightly cron, update the settings for a given repository.
  */
 export function handler(app: Application) {
+
+  // Lint any pull requests that touch configuration
+  app.on(
+    [
+      'pull_request.opened',
+      'pull_request.reopened',
+      'pull_request.synchronize',
+    ],
+    async (context: Context) => {
+      const owner = context.payload.repository.owner.login;
+      const repo = context.payload.repository.name;
+      const number = context.payload.number;
+      const files = await context.github.paginate(
+        context.github.pulls.listFiles.endpoint.merge({
+          owner,
+          repo,
+          pull_number: number,
+          per_page: 100,
+        })
+      );
+      for (const file of files) {
+        if (
+          file.status === 'deleted' ||
+          (file.filename !== `.github/${configFileName}` &&
+          (repo !== '.github' || file.filename !== configFileName))
+        ) {
+          continue;
+        }
+        const blob = await context.github.git.getBlob({
+          owner,
+          repo,
+          file_sha: file.sha,
+        });
+        const configYaml = Buffer.from(blob.data.content, 'base64').toString('utf8');
+        const config = yaml.safeLoad(configYaml, {
+
+        });
+        let isValid = false;
+        let errorText = '';
+        if (typeof config === 'object') {
+          const validateSchema = ajv.compile(schema);
```

Comment:
this could be a nice approach to use in some other places, for instance when  providing configuration for CHANGELOG generation in release-please.

Reply:
I want to get this all happily working, then I'd like to up-level the support to `gcf-utils`.  There are an annoying number of moving parts here, and having a generic utility feels nice.

Reply:
+1 -- this would be great for other bots. I'm very tempted to copy it as-is.